w12h02/test/pgdp/ds/MultiStackTest.java

```diff
@@ -0,0 +1,305 @@
+    }
+
+    @Test
+    void pushShouldNotChangeHead() {
+        final var head = getHead();
+        stack.push(200);
+        stack.push(200);
+        stack.push(200);
+        assertEquals(head, getHead());
+    }
+
+    @Test
+    void topShouldReturnCorrectValues() {
+        final var value = 10;
+        final var value2 = 20;
+        final var value3 = 30;
+        final var value4 = 40;
+        
+        assertEquals(Integer.MIN_VALUE, stack.top());
+        stack.push(value);
+        assertEquals(value, stack.top());
+        stack.push(value2);
+        assertEquals(value2, stack.top());
+        stack.push(value3);
+        assertEquals(value3, stack.top());
+        stack.push(value4);
+        assertEquals(value4, stack.top());
+
+        assertArrayEquals(array(value), getMem(getHead()));
+        assertArrayEquals(array(value2, value3), getMem(getNext(getHead())));
+        assertEquals(value4, getMem(getNext(getNext(getHead())))[0]);
+    }
+
+
+    @Test
+    void popShouldReturnCorrectValues() {
+        final var value = 10;
+        final var value2 = 20;
+        final var value3 = 30;
+        final var value4 = 40;
+
+        stack.push(value);
+        assertEquals(value, stack.pop());
+
+        stack.push(value2);
+        stack.push(value3);
+        assertEquals(value3, stack.pop());
+
+        assertArrayEquals(array(value2), getMem(getHead()));
+
+        stack.push(value4);
+        stack.push(value3);
+        assertEquals(value3, stack.pop());
+        assertEquals(value4, stack.pop());
+        assertEquals(value2, stack.pop());
+
+        assertEquals(Integer.MIN_VALUE, stack.pop());
+    }
+
+    @Test
+    void shouldNotCreateNewStackWhenCurrentStackIsNotFull() {
+        stack.push(200);
+        stack.push(200);
+        stack.push(200);
+        stack.pop();
+        stack.push(200);
+        final var next = getNext(getHead());
+        assertNull(getNext(next));
+    }
+
+    @Test
+    void multiThreadShouldContainAllValues() throws InterruptedException {
+        int[] values = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+        int[] values1 = {10, 11, 12, 13, 14, 15, 16, 17, 18, 19};
+        List<Integer> expected = new ArrayList<>();
+        for (int i = 0; i < values.length; i++) {
+            expected.add(values[i]);
+            expected.add(values1[i]);
+        }
+
+        StackThread thread = new StackThread(stack, values, Task.ContainAllValues);
+        StackThread thread1 = new StackThread(stack, values1, Task.ContainAllValues);
+        thread.start();
+        thread1.start();
+        //noinspection ResultOfMethodCallIgnored
+        stack.toString();
+        stack.top();
+        stack.size();
+        thread.join();
+        thread1.join();
+
+        assertEquals(expected.size(), stack.size());
+
+        for (int i = 0; i < expected.size(); i++) {
+            int value = stack.pop();
+            assertTrue(expected.contains(value));
+            expected.remove(value);
```

Comment:
I think here you wanted to use the `remove(Object o)` method, right? To do this, you have to cast the value to an integer object. Otherwise, the index at value is removed.